benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java

```diff
@@ -381,7 +395,7 @@ private TestResult generateLoad(List<BenchmarkProducer> producers, long testDura
                             rateLimiter.acquire();
 
                             final long sendTime = System.nanoTime();
-                            String key = randomKeys[Math.abs(((int) sendTime)) % randomKeys.length];
+                            String key = keyDistributor.next();
```

Comment:
Since the key distributor is not thread-safe, we should use one instance for each producing thread.

Reply:
nice catch